pkg/utils/dyn_pprof_unix.go

```diff
@@ -17,16 +17,16 @@ const startPProfSignal = syscall.SIGUSR1
 var signalChan = make(chan os.Signal, 1)
 
 // StartDynamicPProfListener starts the listener that will enable pprof when received `startPProfSignal`.
-func StartDynamicPProfListener() {
+func StartDynamicPProfListener(start func(addr string)) {
```

Comment:
Prefer pass arguments explicitly instead of a closure.

Reply:
This function is an internal function, I think if we pass arguments here, we probably need to provide TLS config explicitly here, like `StartDynamicPProfListener(cert, key string)`. At the implementation might would be like:

```go
if cert == "" || key == "" {
  StartPProfListener("0.0.0.0:0")
} else {
  StartPProfListener("0.0.0.0:0", cert, key)
}
```

We got more args, and function overloading-like magic... Without more information, I'm afraid I would prefer the current style...

Reply:
We can move the if condition into the `StartPProfListener`.

```go
func StartPProfListener(addr, cert, key string) error {
    if cert == "" || key == "" {
    } else {
    }
}
```

Reply:
Yep, I think those would make a `if` statement within the function, whose condition is the new parameter we added. And in my opinion this is also control coupling or another form of function overloading. 

I think make another function and let user to decide how to start the pprof listener would be more flexible. 

I'm wondering if there are some style guides preferring plain arguments or some bad things happened with closures...? If so, I will surely remove those.